### PR TITLE
refactor(text-input): remove tag styles for inputs

### PR DESCRIPTION
**Pull request by priyajeet**
Does not fix other components using text-inputs or text-areas

BREAKING CHANGE: The styling classes for text inputs have changed.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @priyajeet has signed the CLA. Thanks for the pull request!

File: src/components/text-input/TextInput.js [hunk 1]
```diff
@@ -46,29 +45,28 @@ const TextInput = ({
     labelTooltip,
     ...rest
 }: Props) => {
-    const classes = classNames(className, 'text-input-container', {
-        'show-error': !!error,
+    const classes = classNames(className, 'bdl-TextInput-container', {
+        'bdl-hasError': !!error,
```
Comment (priyajeet) [inline src/components/text-input/TextInput.js]: No TextInput- for this as this is a state that other places can re-use. Downside of this is that people should only have it `&`ed and never styled on its own.

File: src/components/text-input/TextInput.scss [hunk 2]
```diff
@@ -1,55 +1,105 @@
...
+@import '../../styles/inputs';
 
-/**************************************
- * Text Input
- **************************************/
+.bdl-TextInput-input {
+    @include common-typography;
+
+    height: 32px;
+    margin-top: 5px;
```
Comment (priyajeet) [inline src/components/text-input/TextInput.scss]: Margin should likely not be there, the component should not touch its outside

File: src/components/text-input/TextInput.scss [hunk 3]
```diff
@@ -1,55 +1,105 @@
 @import '../../styles/variables';
+@import '../../styles/inputs';
 
-/**************************************
- * Text Input
- **************************************/
+.bdl-TextInput-input {
```
Comment (priyajeet) [inline src/components/text-input/TextInput.scss]: This is not scoped inside `.bdl-TextInput-container` so that this class can be used on its own if needed. For some strangling html `<input>`.

File: src/components/text-input/TextInput.scss [hunk 4]
```diff
@@ -1,55 +1,105 @@
...
+    @include common-typography;
```
Comment (priyajeet) [inline src/components/text-input/TextInput.scss]: For similar reasons of not being nested inside a container, and also usefulness inside a tethered component

File: src/styles/common/_typography.scss [hunk 5]
```diff
@@ -110,38 +109,3 @@ textarea {
...
-
-.txt-medium {
-    font-size: 14px;
-}
-
-.txt-large {
-    font-size: 16px;
-}
-
-.txt-x-large {
```
Comment (priyajeet) [inline src/styles/common/_typography.scss]: Should not be using all this stuff

File: src/styles/common/_forms.scss [hunk 6]
```diff
@@ -4,144 +4,6 @@
...
-    vertical-align: top;
-}
-
-.input-x-small,
-.input-x-small[type] {
-    width: 100px;
-}
-
-.input-small,
-.input-small[type] {
```
Comment (priyajeet) [inline src/styles/common/_forms.scss]: Should not be using all this stuff

File: src/components/text-input/TextInput.scss [hunk 7]
```diff
@@ -1,55 +1,105 @@
...
+    &[type="email"],
+    &[type="url"],
+    &[type="tel"],
+    &[type="number"] {
+        @include box-inputs;
+
+        width: 262px;
+
+        &:hover {
+            @include box-inputs-hover;
```
Comment (priyajeet) [inline src/components/text-input/TextInput.scss]: @includes will be coming from _textInputs.scss, so these will be text-input-styles, text-input-hover-styles etc.

File: src/components/text-input/TextInput.js [hunk 8]
```diff
@@ -46,29 +45,28 @@ const TextInput = ({
...
                 hideLabel={hideLabel}
                 showOptionalText={!hideOptionalLabel && !isRequired}
                 text={label}
                 tooltip={labelTooltip}
             >
-                {!!description && <i className="text-input-description">{description}</i>}
+                {!!description && <i className="bdl-TextInput-description">{description}</i>}
                 <Tooltip isShown={!!error} position={errorPosition || 'middle-right'} text={error || ''} theme="error">
-                    <input ref={inputRef} required={isRequired} {...rest} />
+                    <input {...rest} className="bdl-TextInput-input" ref={inputRef} required={isRequired} />
```
Comment (jstoffan) [inline src/components/text-input/TextInput.js]: Moving `{...rest}` seems like a risky-ish change. Moving it doesn't seem related to styling, since the class names are stripped out in the method signature.

File: src/components/label/StandardLabel.js [hunk 9]
```diff
@@ -5,13 +5,19 @@ import Tooltip from '../tooltip';
 import LabelPrimitive from './LabelPrimitive';
 
 type Props = {
+    /** Add a class to the component */
```
Comment (jstoffan) [inline src/components/label/StandardLabel.js]: This looks mispositioned.

File: src/components/text-input/TextInput.scss (hunk 3 above)
Comment (jstoffan) [inline src/components/text-input/TextInput.scss]: Are we using `.bdl-` as the new standard prefix for common styles? Are we using BEM going forward, as well, with `TextInput` being BookCase?

File: src/styles/common/_forms.scss [hunk 10]
```diff
@@ -4,144 +4,6 @@
...
-
-input[type="text"].is-invalid,
-input[type="password"].is-invalid,
-input[type="search"].is-invalid,
-input[type="email"].is-invalid,
-input[type="url"].is-invalid,
-input[type="tel"].is-invalid,
-input[type="number"].is-invalid,
-div[contentEditable="true"].is-invalid,
-textarea.is-invalid {
```
Comment (priyajeet) [inline src/styles/common/_forms.scss]: Dont think is-invalid is used

File: src/components/text-input/TextInput.js [hunk 11]
```diff
@@ -46,29 +45,28 @@ const TextInput = ({
...
-    const classes = classNames(className, 'text-input-container', {
-        'show-error': !!error,
+    const classes = classNames(className, 'bdl-TextInput-container', {
+        'bdl-hasError': !!error,
     });
 
     return (
         <div className={classes}>
             <Label
+                className="bdl-TextInput-label"
```
Comment (priyajeet) [inline src/components/text-input/TextInput.js]: This PR is using https://github.com/suitcss/suit/blob/master/doc/naming-conventions.md as naming example

File: src/components/text-input/TextInput.scss (hunk 3 above)
Comment (priyajeet) [reply src/components/text-input/TextInput.scss]: This example is following SUIT css style. Still TBD if we will use SUIT vs BEM.

File: src/components/text-input/TextInput.scss [hunk 12]
```diff
@@ -1,55 +1,105 @@
...
     position: relative;
 
-    .text-input-loading {
+    &.bdl-hasError {
+        .bdl-TextInput-input {
+            border-color: $amaranth;
+        }
+    }
+
+    .bdl-TextInput-loadingIndicator {
```
Comment (priyajeet) [inline src/components/text-input/TextInput.scss]: All these `bdl-TextInput-FOO` styles here can/should be replaced with SCSS convention `&-FOO`

**Comment by priyajeet** [comment]
Linting for this is a bit annoynig with poor examples/docs. But roughly using postcss-bem-linter the rules are like

```js
const customSuitSelector = component => {
    const namespace = 'bdl';
    const block = component; // [A-Z]+
    const camelCase = '[a-z][a-zA-Z0-9]*';
    const state = `(?:.${namespace}-(is|has)-${camelCase})?`;
    const element = `(?:-${camelCase})?`;
    const modifier = `(?:--${camelCase})?`;
    const attribute = '(?:\\[.+\\])?';
    return new RegExp(`^\\.${namespace}-${block}${element}${modifier}${attribute}${state}$`);
};
```

File: src/components/text-input/TextInput.scss (hunk 12 above)
Comment (booc0mtaco) [reply src/components/text-input/TextInput.scss]: How far would we want to go with this? For example, `${namespace}-${ComponentName}` is the top-level selector name? or just `${namespace}` with `-${ComponentName}` immediately after as a nested partial selector?

Comment (priyajeet) [reply src/components/text-input/TextInput.scss]: Based on some feedback we might as well just keep full strings for searchability. Since concatenation can make that harder

**Comment by priyajeet** [comment]
I busted my master branch, so had to open a diff draft PR for this
